### PR TITLE
feat(API): add optionSelected property

### LINKED ISSUE #92: Menu needs to store the selected item as an object that can be queried when selectedOption event fires
Currently we only pass `value`. We should pass `displayValue` as well. Without it parent components have to do a lot of work to figure out what the displayValue should be and this could be fragile.

Related to AlaskaAirlines/auro-select#40 

Exit Criteria:
selectedOption event no longer passes the value
During a select the entire object selected is stored to a property that is public

### LINKED ISSUE #95: Invalid use of `aria-selected`
## Describe the bug

The `aria-selected` attribute is not valid when using `role="menu"` and `role="menuitem"`.

This is outlined here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected

Changing the roles to `tab` and `tablist` makes the `aria-selected` attribute compliant.

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #92

## Summary:

This change adds a new `optionSelected` property. This will be used as a quick efficient means to get all data associated with the currently selected menu option.

Currently `auro-select` will loop through all menu-options looking for one that matches the saved value and then fetch the data it needs off of that object. With this change select will simply be able to do `auro-menu.optionSelected.innerText` or any other normal query directly.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/auro-menu.js [hunk 1]
```diff
@@ -124,61 +162,21 @@ class AuroMenu extends LitElement {
...
-      }
-    }));
-  }
-
   /**
    * Used for @slotchange event on slotted element.
    * @private
    */
   handleSlotItems() {
+    this.setAttribute('role', 'tablist');
```
Comment (blackfalcon) [inline src/auro-menu.js]:
```suggestion
    this.setAttribute('role', 'menu');
```

File: src/auro-menuoption.js [hunk 2]
```diff
@@ -58,6 +58,10 @@ class AuroMenuOption extends LitElement {
     ];
   }
 
+  firstUpdated() {
+    this.setAttribute('role', 'tab');
```
Comment (blackfalcon) [inline src/auro-menuoption.js]:
```suggestion
    this.setAttribute('role', 'menuitem');
```

File: src/auro-menu.js [hunk 3]
```diff
@@ -39,31 +41,59 @@ class AuroMenu extends LitElement {
 
   firstUpdated() {
     this.addEventListener('keydown', this.handleKeyDown);
-    this.addEventListener('mousedown', this.handleClick);
-    this.addEventListener('click', this.selectedOption);
+    this.addEventListener('click', this.makeSelection);
```
Comment (blackfalcon) [inline src/auro-menu.js]: Using `click` only checks the option when the mouse click is up. This was discussed with the last iteration. 

```suggestion
    this.addEventListener('mousedown', this.makeSelection);
```

File: src/auro-menu.js [hunk 4]
```diff
@@ -39,31 +41,59 @@ class AuroMenu extends LitElement {
...
-    this.index = this.items.indexOf(event.target);
+  /**
+   * Set the attributes on the selected menuoption, the menu value and stored option.
+   * @param {Object} option - The menuoption to be selected.
+   * @private
+   */
+  handleLocalSelectState(option) {
+    option.tabIndex = 0;
+    option.setAttribute('selected', '');
+    option.ariaSelected = true;
```
Comment (blackfalcon) [inline src/auro-menu.js]: Is the intention with the first load that the first element be selected if there isn't a defined pre-selected option? That's not a spec that I am aware of?

**Review by blackfalcon — CHANGES_REQUESTED** [review]
I guess I wasn't 100% understanding of the exit criteria. When it was explained that an object would be returned, I assumed that this was an object of data. When I a looking at the return of this object, I am given the HTML for the returned element. Is that correct? 

Screen Shot 2022-03-08 at 9 50 32 AM

I feel that there should be additional documentation to support this in the demo or the API doc.

File: src/auro-menu.js (hunk 1 above)
Comment (jason-capsule42) [reply src/auro-menu.js]: role `menu` is incompatible with `aria-selected="true"`. Either we need to ditch aria-selected or we need to change to a role combination that is supported.

This is outlined in the attached issues.

File: src/auro-menuoption.js (hunk 2 above)
Comment (jason-capsule42) [reply src/auro-menuoption.js]: See above comment about aria-selected.

File: src/auro-menu.js (hunk 4 above)
Comment (jason-capsule42) [reply src/auro-menu.js]: This is what was already happening in the existing code.

File: src/auro-menu.js (hunk 1 above)
Comment (geoffrich) [reply src/auro-menu.js]: I don't think `tablist` is an appropriate role here:

> The tablist role identifies the element that serves as the container for a set of tabs. The tab content are referred to as tabpanel elements.

Maybe this should be a [listbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role) with [options](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/option_role)?

Comment (blackfalcon) [reply src/auro-menu.js]: Yes, `listbox` and `option` seem like the right move here versus `tab` update.

File: src/auro-menu.js (hunk 4 above)
Comment (blackfalcon) [reply src/auro-menu.js]: The previous version was only selecting the first option on an initial click event in a special case with an attribute. The current `main` branch does not do this on render. 

https://auro.alaskaair.com/components/auro/menu

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**